Lib/dataclasses.py

```diff
@@ -753,6 +753,8 @@ def _get_field(cls, a_name, a_type):
 def _set_new_attribute(cls, name, value):
     # Never overwrites an existing attribute.  Returns True if the
     # attribute already exists.
+    if isinstance(value, FunctionType):
```

Comment:
Please move these two lines below the `return True` line. The comment above refers to the test and return statement.

Also, please add a comment about why `__qualname__` needs to be set for functions.

Reply:
Didn't notice that all, thanks fixed!